internal/datasource/config/config_writer.go

```diff
@@ -36,46 +37,40 @@ type (
 		Complete() (err error)
 	}
 
-	Client struct {
-		Timeout time.Duration
-	}
-
-	ConfigWriterParameters struct {
-		configClient client.HTTPConfigClientInterface
-		Client       Client
-		cachePath    string
-	}
-
 	ConfigWriter struct {
 		configClient       client.HTTPConfigClientInterface
 		previouseFileCache FileCache
 		currentFileCache   FileCache
 		cachePath          string
 		dataplaneConfig    config.DataplaneConfig
+		allowedDirectories []string
 	}
 
 	// map of files with filepath as key
 	FileCache = map[string]*instances.File
 )
 
-func NewConfigWriter(configWriterParameters *ConfigWriterParameters, instanceID string) *ConfigWriter {
-	if configWriterParameters.cachePath == "" {
-		configWriterParameters.cachePath = fmt.Sprintf(cacheLocation, instanceID)
+func NewConfigWriter(configClient client.HTTPConfigClientInterface, cachePath string,
```

Comment:
Do you need to pass the cachePath in here as a parameter? Could you create the cachePath in the function using the instanceID?

Reply:
Ya I can, I wasn't sure what we were doing with the cache path and if it would always be at that location so left it as a parameter. I'll remove it